src/ansi-c/c_expr.h

```diff
@@ -225,4 +225,69 @@ to_history_expr(const exprt &expr, const irep_idt &id)
   return ret;
 }
 
+/// \brief A class for an expression that represents a
+/// conditional target in an assigns clause.
+class conditional_assigns_target_exprt : public binary_exprt
+{
+public:
+  explicit conditional_assigns_target_exprt(exprt _condition, exprt _target)
+    : binary_exprt(
+        std::move(_condition),
+        ID_conditional_assigns_target,
+        std::move(_target))
```

Comment:
Nit pick: using this constructor of `binary_exprt`, the expression will implicitly have as `type()` whatever `_condition.type()` is. We shouldn't really use the type of this expression, so I'd suggest to explicitly set the type either as `typet{}` (and use the 4-parameter constructor of `binary_exprt`) to make sure no one ends up using that type.